src/ert/storage/local_experiment.py

```diff
@@ -269,6 +269,17 @@ def metadata(self) -> dict[str, Any]:
         with open(path, encoding="utf-8") as f:
             return json.load(f)
 
+    @require_write
+    def update_metadata(self, data: dict[str, Any]) -> None:
+        """
+        Update the metadata file for the experiment.
+        """
+        path = self.mount_point / self._metadata_file
+        assert path.exists()
```

Comment:
If this happens, do we really want an AssertionError or would we rather like an OSError which will be thrown in  `_write_transaction` below?